### PR TITLE
UI change of minimized programming instructions header to look more like problem statement header

### PR DESCRIPTION
**Pull request by Malyuk-A**
### Checklist
- [x] We tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: We followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] Client: We added multiple screenshots/screencasts of my UI changes
- [x] Client: We translated all newly inserted strings into English and German.

### Motivation and Context

The first couple of images show the current UI for instructions (i.e. problem statement for programming exercises).
If you click on the header you can minimize the instructions to give the code editor more space.

programming : normal view [old]            |  programming : collapsed view [old]
:-------------------------:|:-------------------------:
  | 

If you compare this UI with the other exercise types you can spot that the design is quite different therefore it breaks the homogeneous look and feel of the exam mode.

other exercises : normal view  | other exercises : collapsed view
:-------------------------:|:-------------------------:
 | 

This issue has been spotted and requested to be fixed during the developer meeting's exam testing session (13.07.2021).

### Description

We changed the instructions header to resemble the other exercise type's problem statement as close as possible.
<details>
<summary> remarks about the code structure </summary>
<p> --------------------------------------------------- </p>
<p>At first, we looked at the implementation of the problem statement header for text exercises and tried to reuse it for programming exercises. </p>
<p>The problem statement itself is located in `text-exam-submission.component.html` which itself is embedded in `resizeable-container.component.html` (containers are connected to CSS grids that determine the location of elements).</p>
<p>The logic behind the collapsed/minimized and open/normal view is based on a simple `*ngIf="collapsed; then collapsedBody; else expandedBody` rule. I.e. a private component property `collapsed` determines what kind of element should be shown/created. </p>
This reasoning is quite easy, and we thought that a similar logic will be used for programming exercises.

Programming exercises' instructions are located inside `programming-exam-submission.component.html` (right next to the text exercise's one when you look at the file hierarchy, i.e. inside **../exam/participate/exercises**)
But the logic is sadly way more complex, coupled, and confusing to understand. 
Multiple different TS, CSS, HTML files that are all interconnected => high coupling
It is quite hard to understand what influences what and how.
Instead of a simple *ngIf case the collapsing is done via event handling and grid methods that resize the visible element.
<p> => this code was mostly written in 2019 and really needs refactoring </p>

<p> --------------------------------------------------- </p>
</details>
We changed the header text from "Instructions" to "Problem Statement" (including German translation).
_The other (visual) differences can be seen more easily than described by words._

### Result
programming : normal view [new]            |  programming : collapsed view [new]
:-------------------------:|:-------------------------:
  | 

The new problem statement for programming exercises still has two minor differences from the other exercise types:
- the font size is a bit larger for programming exercises, we decided to keep it because otherwise, it will not fit with the other texts on the online code editor.
- there is a visible icon "||" that is used for changing the size of the opened window when dragged. We decided to keep this because it also stays when the other collapsable elements are minimized (e.g. file browser). 

### Steps for Testing

1. Log in to Artemis
2.a Go to an exam with a programming exercise (create one if you can not find one)
2.b Go to a normal course programming exercise and click on online code editor (it uses the same UI)
3. Look at the instructions (header) : it has to look like the old one
4. Click on it and make sure it minimizes correctly (and opens back up successfully as well)
5. The minimized instructions should look more in tune with the other exercise types (problem statement header)
6. Make sure the final result looks like our screenshots.

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/shared/code-editor/container/code-editor-container.component.html [hunk 1]
```diff
@@ -49,9 +49,11 @@ <h4><ng-content select="[editorTitle]"></ng-content></h4>
         (onFileLoad)="fileLoad($event)"
     >
     </jhi-code-editor-ace>
+
```
Comment (SimonEntholzer) [inline src/main/webapp/app/exercises/programming/shared/code-editor/container/code-editor-container.component.html:52]:
```suggestion
```

File: src/main/webapp/app/exercises/programming/shared/code-editor/container/code-editor-container.component.html [hunk 2]
```diff
@@ -49,9 +49,11 @@ <h4><ng-content select="[editorTitle]"></ng-content></h4>
...
     <jhi-code-editor-instructions *ngIf="showEditorInstructions" editorSidebarRight (onToggleCollapse)="onToggleCollapse($event)" (onError)="onError($event)">
         <ng-content select="[editorSidebar]"></ng-content>
     </jhi-code-editor-instructions>
+
```
Comment (SimonEntholzer) [inline src/main/webapp/app/exercises/programming/shared/code-editor/container/code-editor-container.component.html:56]:
```suggestion
```

File: src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.scss [hunk 3]
```diff
@@ -186,7 +198,7 @@ Collapsable styling
         grid-template-columns: minmax(max-content, 1fr) minmax(max-content, 2fr);
 
         > :not(:first-child) {
-            border-left: 1px solid rgba(0, 0, 0, 0.125);
+            border-left: 3px solid;
```
Comment (SimonEntholzer) [inline src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.scss:201]:
```suggestion
            border-left: 1px solid rgba(0, 0, 0, 0.125);
```

**Review by rs-maker — DISMISSED** [review]
Tested on ts3 works great.

**Review by FrankeLukas — DISMISSED** [review]
Lgtm.

**Review by MDK5 — CHANGES_REQUESTED** [review]
Tested on ts1, style for exams is unified now.

However the style for assessment is now different:
Other (e.g. text):
grafik
Other (e.g. modeling):
grafik
Programming:
grafik

I'm afraid the code editor has to use different texts for each usage.

Also, but maybe out of scope: modeling exercises have the text in bold and with different capitalization for some reason:
Other:
grafik
Modeling:
grafik

**Review by argertboja — DISMISSED** [review]
Tested on TS1 and LGTM 👍🏼

**Review by canyelten — DISMISSED** [review]
Tested on TS1, works as expected.

**Comment by krusche** [comment]
> Tested on ts1, style for exams is unified now.
> 
> However the style for assessment is now different:
> Other (e.g. text):
> grafik
> Other (e.g. modeling):
> grafik
> Programming:
> grafik
> 
> I'm afraid the code editor has to use different texts for each usage.
> 
> Also, but maybe out of scope: modeling exercises have the text in bold and with different capitalization for some reason:
> Other:
> grafik
> Modeling:
> grafik

I think we can fix this issue in a follow-up PR

**Review by Olimbia23 — APPROVED** [review]
Tested it on TS2.
The result looks as in the screenshots. Nice work @Malyuk-A

**Review by ArtemTsurikov — APPROVED** [review]
Tested on TS1, looks good

**Review by FrankeLukas — APPROVED** [review]
Re-approve (only style & test fixes).

**Review by maximilianbergmann — APPROVED** [review]
Tested on TS3

**Review by real-dmy — APPROVED** [review]
Tested on TS1, works as expected.

**Review by katjanakosic — APPROVED** [review]
Looks good on ts1

**Review by sinangunduz — APPROVED** [review]
tested on ts2, works as expected

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Looks good to me!

**Review by NicolasUllmann — APPROVED** [review]
Tested on TS1. Everything worked as expected.

**Comment by Malyuk-A** [comment]
This PR is part of https://github.com/ls1intum/Artemis/pull/3832
therefore the code was already merged.
I will close this PR

**Comment by Malyuk-A** [comment]
This PR is part of https://github.com/ls1intum/Artemis/pull/3832
therefore the code was already merged.
I will close this PR

**Pull request closed without merging**